downloads/downloads-impl/src/main/java/com/duckduckgo/downloads/impl/DefaultDownloadsRepository.kt

```diff
@@ -44,21 +47,33 @@ class DefaultDownloadsRepository @Inject constructor(
 
     override suspend fun update(downloadId: Long, downloadStatus: Int, contentLength: Long) {
         downloadsDatabase.downloadsDao().update(downloadId, downloadStatus, contentLength)
+        if (downloadStatus != DownloadStatus.STARTED) {
+            urlFileDownloadCallManager.remove(downloadId)
+        }
     }
 
     override suspend fun update(fileName: String, downloadStatus: Int, contentLength: Long) {
         downloadsDatabase.downloadsDao().update(fileName, downloadStatus, contentLength)
     }
 
-    override suspend fun delete(id: Long) {
-        downloadsDatabase.downloadsDao().delete(id)
+    override suspend fun delete(downloadId: Long) {
```

Comment:
Rename `downloadId` to `id` to match the declaration in the interface.

Reply:
will do